### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by BerradaYassine**

### CODE REVIEW

**Comment by eugen-natucci** [comment]
@mhmdio Hi Mohammed,

can you please advise on what would be the best approach here in Yasssine's PR:

because he needs one part to be executed before the other one (issue and validate cert and then create frontend bucket with all the rest) he splits his work in two directories: one part goes to the root dir and the rest inside `front-end` dir. And because of that we'll be applying the infrastructure in two steps.

@BerradaYassine please correct me if I missed or misinterpreted anything

**Comment by BerradaYassine** [comment]
Yes, first part

deploying the root module including the creation of the hosted zone (
because i will need them to manage the hosted zone records first at ease
for the ssl certificate to issue ) - otherwise they would have to do the
management of their hosted zone records quickly or else the front end
deployement will crash if they don't donit quickly ) so I seperated them as
first creating the hosted zone and managing once that is done

deploying the front end



Yours,
Yassine

Le mer. 17 juin 2020 à 12:57, Eugen Natucci <notifications@github.com> a
écrit :

**Comment by mhmdio** [comment]
I see, I suggest to use some sort of make file style, I used https://taskfile.dev/#/ to create Terraform workflow, and you can import outputs from first part state file using TF state file data sources.

**Comment by mhmdio** [comment]
Sample task file:

```yml
version: '2'

env:
  TERM: screen-256color
  GO111MODULE: on
  AWS_DEFAULT_REGION: ca-central-1

tasks:
  default:
    cmds:
    - task: test

  test:
    desc: Run tests.
    cmds:
    - task: test-terraform
    - task: tflint-terraform

  test-terraform:
    desc: Run tests for all terraform directories.
    silent: true
    env:
      DIRECTORIES:
        sh: find . -type f -name '*.tf' -not -path "**/.terraform/*" -print0 | xargs -0I {} dirname {} | sort -u
    cmds:
    - |
      BOLD=$(tput bold)
      NORM=$(tput sgr0)
      CWD=$PWD
      for d in $DIRECTORIES; do 
        cd $d
        echo "${BOLD}$PWD:${NORM}"
        if ! terraform fmt -check=true -list=false -recursive=false; then
          echo "  ✗ terraform fmt" && exit $?
        else
          echo "  √ terraform fmt"
        fi
        if ! terraform init -backend=false -input=false -get=true -get-plugins=true -no-color > /dev/null; then
          echo "  ✗ terraform init" && exit $?
        else
          echo "  √ terraform init"
        fi
        if ! terraform validate > /dev/null; then
          echo "  ✗ terraform validate" && exit $?
        else
          echo "  √ terraform validate"
        fi
        cd $CWD
      done

  tflint-terraform:
    desc: Run tflint for simple example.
    silent: true
    cmds:
    - | 
      cd examples/simple 
      tflint --deep
      echo "  √ tflint --deep"
```

**Comment by BerradaYassine** [comment]
I see, yes it sounds good I've never worked with it before but i'll be on it thank you ! Also for the workflow to work

**Comment by BerradaYassine** [comment]
I see, yes it sounds good to me I've never used it before but i'll be on it thank you ! also just to be sure , i should still keep the front-end sub module right ?

**Comment by mhmdio** [comment]
I think there are a couple of ways to do it:
- keep it as a submodule and create a workflow
- move it to the root level and have it beside current modules, and creates a workflow
- if you use TF cloud you can create triggers between 1st templates and then 2nd templates
- TF 0.13 will support module depends_on.
- use terragrunt https://terragrunt.gruntwork.io/docs/features/execute-terraform-commands-on-multiple-modules-at-once/

It depends on what you use now to apply TF?

File: variables.tf [hunk 1]
```diff
@@ -1,7 +1,35 @@
 variable "acdb-user" {
     type = string
+    default = "acdbuser"
 }
 
 variable "acdb-password" {
```
Comment (mhmdio) [inline variables.tf:6]: You can get this value from Secretsmanager

File: variables.tf [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
     type = string
+    default = "password"
 }
+
+variable "ac-backend-jwt-secret" {
```
Comment (mhmdio) [inline variables.tf:11]: You can get this value from Secretsmanager

File: variables.tf [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+    type = string
+    default = "-"
+}
+
+variable "ac-backend-db-url" {
```
Comment (mhmdio) [inline variables.tf:16]: You can save this in SSM Parameter store, and use datasource to read it

File: variables.tf (hunk 2 above)
Comment (eugen-natucci) [reply variables.tf:11]: Yes, they go to SecretManager in secrets.tf. But they need to get there somehow from the beginning. Same for parameters  ssm-parameters.tf…

**Comment by eugen-natucci** [comment]
I squashed all the commits here into one and cherry-picked into master: e03ab44a5402009e476d77d74c51569815bb936a. So all your changes @BerradaYassine are in master, I'm rejecting this pull reuqest.

**Pull request closed without merging**